src/Services/GTT/shared/GTT.Infrastructure/Repositories/ChallengeRepository.cs

```diff
@@ -1,72 +1,75 @@
-
-            return result;
+            _connection = dbConnection;
         }
 
-        public async Task<IReadOnlyList<Challenge>> GetAllAsync()
+        public async Task<BaseResponseModel> AddAsync(CreateChallengeData challenge)
         {
-            var insertChallengeSql = @"SELECT * FROM Challenge";
-            var result = await _connection.QueryAsync<Challenge>(insertChallengeSql, _tran);
+            try
+            {
+                var checkclass = await checkClassExist(challenge.ClassID);
+                if(!checkclass)
+                {
+                    return new BaseResponseModel(HttpStatusCode.NotFound, "Class ID invalid");
+                }
 
-            return result.ToList();
-        }
+                var insertChallengeSql = @"INSERT INTO Challenge(Calories, SplatPoints, AvgHR, MaxHR, Miles, Steps, ClassID, CreatedDate, UpdatedDate, CreatedBy, UpdatedBy)
+                                    VALUES(@Calories, @SplatPoints, @AvgHR, @MaxHR, @Miles, @Steps, @ClassID, @CreatedDate, @UpdatedDate, @CreatedBy, @UpdatedBy)
+                                    DECLARE @challengeID int
+                                    SET @challengeID = SCOPE_IDENTITY()
+                                    SELECT* FROM Challenge WHERE challengeID = @ChallengeId
+                                    ";
 
-        public async Task<Challenge> GetByIdAsync(int id)
-        {
-            var insertChallengeSql = @"SELECT * FROM Challenge WHERE Id = @Id";
-            var result = await _connection.QuerySingleOrDefaultAsync<Challenge>(insertChallengeSql, new { Id = id }, _tran);
+                var param = new
+                {
+                    Calories = challenge.Calories,
+                    SplatPoints = challenge.SplatPoints,
+                    AvgHR = challenge.AvgHr,
+                    MaxHR = challenge.MaxHr,
+                    Miles = challenge.Miles,
+                    Steps = challenge.Steps,
+                    ClassID = challenge.ClassID,
+                    CreatedDate = challenge.CreatedDate,
+                    UpdatedDate = challenge.UpdatedDate,
+                    CreatedBy = challenge.CreatedBy,
+                    UpdatedBy = challenge.UpdatedBy,
+                };
 
-            return result;
+                var result = await _connection.QueryFirstAsync<ChallengeVM>(insertChallengeSql, param);
+                return new BaseResponseModel(result);
+            }
+            catch (Exception ex)
+            {
+                throw new Exception(ex.Message);
+            }     
         }
 
-        public async Task<int> UpdateAsync(Challenge entity)
+        private async Task<bool> checkClassExist(int classid)
         {
-            var sql = @"UPDATE Challenge SET something equal something";
-            var result = await _connection.ExecuteAsync(sql, entity, _tran);
+            var query = @"SELECT * FROM Class WHERE ClassId = @classid";
```

Comment:
Should select ClassId field because we just need to check whether the Class exists